### PR TITLE
Buildvocab

### PR DESCRIPTION
**Pull request by natsegal**
Small bug fixes for vocabulary build  and add the possibility for operators to add extra tokens such as placeholders.

### CODE REVIEW

**Review by guillaumekln — CHANGES_REQUESTED** [review]
Is it possible to add some tests, or do you think the existing ones are enough to check buildvocab?

File: nmtwizard/preprocess/tu.py [hunk 1]
```diff
@@ -120,8 +122,21 @@ def adjust_alignment(self, side_idx, start_idx, tok_num, new_tokens=None, part =
                     new_alignment.add((opp_side_tok_idx, side_tok_idx))
                 else:
                     new_alignment.add((side_tok_idx, opp_side_tok_idx))
+
             self.__alignments[part] = new_alignment
 
+    def insert_alignments(self, src_start_idx, tgt_start_idx, src_nb_inserted_tokens, tgt_nb_inserted_tokens, part = 0):
```
[the remark this answers is not in the document]
Comment (natsegal) [reply nmtwizard/preprocess/tu.py:128]: Yes, I added it for my tests with vocabularies, I'll remove it

File: nmtwizard/preprocess/preprocess.py [hunk 2]
```diff
@@ -20,9 +20,9 @@ def _get_tok_configs(config):
     tok_configs = []
     preprocess_config = config.get("preprocess")
     if preprocess_config is not None:
-        for operator_config in preprocess_config:
+        for i, operator_config in enumerate(preprocess_config):
```
[the remark this answers is not in the document]
Comment (natsegal) [reply nmtwizard/preprocess/preprocess.py:23]: `operator_info_generator` returns a copy of operator parameters, whereas here we manipulate and modify the actual tokenization configuration (add vocabulary and subword module paths).

**Comment by natsegal** [comment]
I did my tests directly with the whole framework and bigger amounts of data, to see what's actually happening (some statistics are difficult to catch with small amounts of data, and placeholders are not added in nmt-wizard-docker) . 
I'll try to think of and add some meaningful unit tests though.

**Pull request merged**